controller/users.go

```diff
@@ -118,7 +118,29 @@ func (c *UsersController) Create(ctx *app.CreateUsersContext) error {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
 	}
 	if userExists {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewVersionConflictError("user with such email or username already exists"))
+		// This may happen for manually deprovisioned users which are reactivating their account via the registration app
+		// We should re-deprovision such user
+		idn, err := c.app.UserService().IdentityByUsernameAndEmail(ctx, ctx.Payload.Data.Attributes.Username, ctx.Payload.Data.Attributes.Email)
```

Comment:
The other `userExistsInDB(..) method could be replaced by this. If an identity is returned, then this retrieval of the user/identity second time is not needed.